src/g_text.c

```diff
@@ -795,7 +795,13 @@ void gatom_key(void *z, t_symbol *keysym, t_floatarg f)
             while (i--)
                 rtext_key(t, '\b', &s_);
             rtext_gettext(t, &buf, &bufsize);
-            text_setto(&x->a_text, x->a_glist, buf, bufsize);
+            t_atom *ap = gatom_getatom(x);
```

Comment:
this is not valid C89 code: please only declare variables at the beginning of a code block

Reply:
like this? https://github.com/pure-data/pure-data/pull/1515/commits/d6f80c6c93b4b73f2db7b8b2c51bef57b061bfa0

Reply:
yes, like this.
though typically you could want to keep variable scope as tight as possible, so the `t_atom*ap` declaration could really go here: https://github.com/pure-data/pure-data/blob/d6f80c6c93b4b73f2db7b8b2c51bef57b061bfa0/src/g_text.c#L790-L791

but anyhow. the C89 problem is fixed by d6f80c6